srtcore/core.cpp

```diff
@@ -5774,13 +5780,25 @@ void *CUDT::tsbpd(void *param)
              */
             HLOGC(tslog.Debug, log << self->CONID() << "tsbpd: no data, scheduling wakeup at ack");
             self->m_bTsbPdAckWakeup = true;
-            THREAD_PAUSED();
-            tsbpd_cc.wait();
-            THREAD_RESUMED();
+
+            bool signaled = false;
+            while (!signaled)
+            {
+                // For safety reasons, do wakeup once per 1s and re-check the flag.
+                THREAD_PAUSED();
+                signaled = tsbpd_cc.wait_for(seconds_from(1));
+                THREAD_RESUMED();
+                if (self->m_bClosing)
+                {
+                    HLOGC(tslog.Debug, log << "tsbpd: IPE? Closing flag set in the meantime of waiting. Exiting");
+                    goto ExitLoops;
+                }
+            }
```

Comment:
Why not just let TSBPD to re-check the receiver buffer even in case of a spurious wake-up?

Reply:
Actually, yes, might be a better idea. OTOH this 1s delay is too much if that deadlock would have happened when closing a socket in the group receiver.